core/src/epicli/data/common/ansible/playbooks/roles/repository/files/download-requirements/centos-7/download-requirements.sh

```diff
@@ -79,30 +79,40 @@ create_directory() {
 disable_repo() {
 	local repo_id="$1"
 
-	if yum repolist enabled | grep --quiet "$repo_id"; then
+	if [[ $(yum repolist enabled | grep --quiet "$repo_id") ]]; then
 		echol "Disabling repository: $repo_id"
 		yum-config-manager --disable "$repo_id" ||
 			exit_with_error "Command failed: yum-config-manager --disable \"$repo_id\""
 	fi
 }
 
-# params: <file_url> <dest_dir>
+# params: <file_url> <dest_dir> <new_filename> # $3 not required
 download_file() {
 	local file_url="$1"
 	local dest_dir="$2"
 
-	local file_name=$(basename "$file_url")
-	local dest_path="$dest_dir/$file_name"
-	local retries=3
-
-	# wget with --timestamping sometimes failes on AWS with ERROR 403: Forbidden
-	# so we remove existing file to overwrite it, to be optimized
-	[[ ! -f $dest_path ]] || remove_file "$dest_path"
+	if [[ ${3-} ]]; then
+		local file_name=$3
+	else
+		local file_name=$(basename "$file_url")
```

Comment:
We are not checking exit code of file_name. Only argument I see to declare it separately is for set -e but in our case it will never happen..(for sure we can be extra paranoic and add all these proposals but it can be considered during role refactoring). So I think changing this variable as readonly will be enough.